src/main/scala/org/constellation/domain/healthcheck/ReconciliationRound.scala

```diff
@@ -0,0 +1,244 @@
+package org.constellation.domain.healthcheck
+
+import cats.effect.concurrent.Ref
+import cats.effect.{Blocker, Clock, Concurrent, ContextShift, Fiber, Timer}
+import cats.syntax.all._
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import io.circe.Codec
+import io.circe.generic.semiauto.deriveCodec
+import org.constellation.infrastructure.p2p.{ClientInterpreter, PeerResponse}
+import org.constellation.p2p.Cluster
+import org.constellation.schema.NodeState.{canActAsJoiningSource, isInitiallyJoining, isInvalidForJoining}
+import org.constellation.schema.consensus.RoundId
+import org.constellation.schema.{Id, NodeState}
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+class ReconciliationRound[F[_]](
+  ownId: Id,
+  cluster: Cluster[F],
+  val startedAtSecond: Long
+)(
+  apiClient: ClientInterpreter[F],
+  unboundedBlocker: Blocker
+)(implicit F: Concurrent[F], CS: ContextShift[F], C: Clock[F], T: Timer[F]) {
+  import ReconciliationRound._
+
+  val logger: SelfAwareStructuredLogger[F] = Slf4jLogger.getLogger[F]
+
+  val retryAttempts = 2
+  val retryAfter: FiniteDuration = 60.seconds
+
+  private[healthcheck] val reconciliationResult: Ref[F, Option[ReconciliationResult]] =
+    Ref.unsafe(None)
+
+//  private[healthcheck] val firstClusterAlignmentResults: Ref[F, Option[Map[Id, List[ClusterAlignmentResult]]]] =
+//    Ref.unsafe(None)
+
+//  private[healthcheck] val lastClusterAlignmentResults: Ref[F, Option[Map[Id, List[ClusterAlignmentResult]]]] =
+//    Ref.unsafe(None)
+
+  def getReconciliationResult(): F[Option[ReconciliationResult]] = reconciliationResult.get
+
+  def start(): F[Fiber[F, Unit]] = F.start(run())
+
+  private def run(
+    attemptsLeft: Int = retryAttempts,
+    firstClusterAlignmentResult: Option[Map[Id, List[ClusterAlignmentResult]]] = None,
+    lastClusterAlignmentResult: Option[Map[Id, List[ClusterAlignmentResult]]] = None
+  ): F[Unit] =
+    for {
+      isFirstRun <- F.delay(attemptsLeft == retryAttempts)
+      isLastRun <- F.delay(attemptsLeft == 0)
```

Comment:
I know the very first cmd must be delayed, but the second can be ```isLastRun = attemptsLeft == 0```. also, the first one is imo pure, so there is no need to delay it.

Reply:
I had a situation once where simple computation seemed to be pure, so I just wrapped it in .pure[F]. But then it didn't execute ;). So maybe I'm to careful. Here it seems like .pure[F] should be enough though.

Reply:
and here @kpudlik